### PR TITLE
Add `machine-account-key` command to `bootstrap` CLI

### PR DESCRIPTION
**Pull request by danuio**
- Adds `bootstrap machine-account-key` which generates a machine account key pair and writes it to the `node-machine-account-key.priv.json` which is used by `bootstrap machine-account` command.

--- 

Note: Targeting `danu/1508/bootstrap-key-command` to show a clearer diff but once the target branch is merged into `feature/epochs`, I'll target the feature branch instead.

### CODE REVIEW

File: cmd/bootstrap/cmd/machine_account_key.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	"github.com/onflow/flow-go/cmd/bootstrap/run"
+	"github.com/onflow/flow-go/model/bootstrap"
+	model "github.com/onflow/flow-go/model/bootstrap"
+	ioutils "github.com/onflow/flow-go/utils/io"
+)
+
+var (
+	flagNodeID string
+)
```
Comment (jordanschalm) [inline cmd/bootstrap/cmd/machine_account_key.go:18]: Is this used?

File: cmd/bootstrap/cmd/machine_account_key.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+func init() {
+	rootCmd.AddCommand(machineAccountKeyCmd)
+
+	machineAccountKeyCmd.Flags().BytesHexVar(&flagMachineSeed, "seed", generateRandomSeed(), fmt.Sprintf("hex encoded machine account seed (min %v bytes)", minSeedBytes))
+}
+
+// machineAccountKeyRun generate a machine account key and writes it to a default file path.
+func machineAccountKeyRun(_ *cobra.Command, _ []string) {
```
Comment (jordanschalm) [inline cmd/bootstrap/cmd/machine_account_key.go:36]: Please add a test for this command following the [example from `key`](https://github.com/onflow/flow-go/pull/722/files#diff-fd992efa53b7e741bbf99ff01fce62a02af4798fc9c5dd075489d461efc6e7dcR42).

We should:
* test that the machine account key file is created when we run the command
* test that an existing machine account key file is not overwritten when we run the command where a file already exists

**Review by jordanschalm — COMMENTED** [review]
This should have tests, otherwise looks good to me.

File: cmd/bootstrap/cmd/machine_account_key.go (hunk 1 above)
Comment (danuio) [reply cmd/bootstrap/cmd/machine_account_key.go:19]: no, not sure why I left that in! thanks

File: cmd/bootstrap/cmd/machine_account_key_test.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	flagAddress = "189.123.123.42:3869"
+
+	hook := zeroLoggerHook{logs: &strings.Builder{}}
+	log = log.Hook(hook)
+
+	// run keys command to generate all keys and bootstrap files
+	keyCmdRun(nil, nil)
+	hook.logs.Reset()
+
+	// require log regex to match
```
Comment (danuio) [inline cmd/bootstrap/cmd/machine_account_key_test.go:40]:
```suggestion
```

File: cmd/bootstrap/cmd/machine_account_key.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not generate machine key")
+	}
+	log.Info().Msg("generated machine account private key")
+
+	// construct object to write to file
+	machineAccountPriv := assembleNodeMachineAccountPriv(machineKey)
+
+	writeJSON(machineAccountKeyPath, machineAccountPriv)
+	log.Info().Str("path", machineAccountKeyPath).Msg("wrote machine account private key")
```
Comment (jordanschalm) [inline cmd/bootstrap/cmd/machine_account_key.go:63]: We need to log the public key here as well. When users run this command their next step will be to copy the public key and input it to Flow Port.
```suggestion
	log.Info().Msgf("machine account public key: %s", machineKey.PublicKey().String())
```

File: cmd/bootstrap/cmd/machine_account_key_test.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+)
+
+func TestMachineAccountKeyFileExists(t *testing.T) {
+	var keyFileExistsRegex = `^machine account private key already exists`
+
+	dirName := strconv.FormatInt(time.Now().UnixNano(), 10)
+	regex := regexp.MustCompile(keyFileExistsRegex)
+
+	// command flags
+	flagOutdir = fmt.Sprintf("/tmp/%s", dirName)
```
Comment (jordanschalm) [inline cmd/bootstrap/cmd/machine_account_key_test.go:29]: We have a utility which creates then tears down a temporary directory we can use here.
https://github.com/onflow/flow-go/blob/ff92e3b5a8edfb80eb797e8896c8c2a0bdea8fcd/utils/unittest/unittest.go#L172

**Review by jordanschalm — APPROVED** [review]
🎸

**Review by arrivets — APPROVED** [review]

**Pull request merged**